src/status_im/contexts/settings/wallet/data_store.cljs

```diff
@@ -53,27 +34,30 @@
   "Updates keypairs' accounts to be fully operable based on specified key-uids and an optional operable condition.
 
   Parameters:
-  - :keypairs (seq): A sequence of keypair maps.
+  - :keypairs (map): A map of keypair key-uid to keypair details.
   - :key-uids-set (set): A set of key-uids that need to be checked and updated.
   - :operable-condition (keyword, optional): The condition that the keypair's accounts' operability must meet to be updated.
     If nil or not provided, the function will update all keypairs' accounts.
 
   Returns:
-  - A new sequence with keypairs updated to be fully operable where the conditions are met."
+  - A new map with keypairs updated to be fully operable where the conditions are met."
   [{:keys [keypairs key-uids-set operable-condition]}]
-  (map (fn [keypair]
-         (if (contains? key-uids-set (:key-uid keypair))
-           (-> keypair
-               (update :accounts #(make-keypairs-accounts-fully-operable % operable-condition))
-               (assoc :lowest-operability :fully))
-           keypair))
-       keypairs))
+  (reduce-kv (fn [acc k keypair]
+               (if (contains? key-uids-set k)
+                 (assoc acc
+                        k
+                        (-> keypair
+                            (update :accounts make-keypairs-accounts-fully-operable operable-condition)
+                            (assoc :lowest-operability :fully)))
+                 (assoc acc k keypair)))
+             {}
+             keypairs))
 
 (defn map-addresses-to-key-uids
   [db addresses]
   (reduce (fn [key-uid-set address]
-            (if-let [account (get-in db [:wallet :accounts address])]
-              (conj key-uid-set (:key-uid account))
+            (if-let [account-key-uid (get-in db [:wallet :accounts address :key-uid])]
+              (conj key-uid-set account-key-uid)
               key-uid-set))
           #{}
           addresses))
```

Comment:
This function already existed, but I'm wondering what it does, seems it returns a set of key-uids. The name is `map-addresses-to-key-uids`. but I don't see the mapping

Reply:
This function receives the addresses of accounts in your wallet and returns the key-uid (key pair uid) of those accounts for updating the keypairs of those accounts. It returns a set as the key-uid can be the same for multiple accounts. It doesn't return an actual map. The function name is a bit confusing, I can update it to `get-keypair-key-uids-from-addresses` or something like that.